src/components/ITSBStore/ITSBSearchHandler.jsx

```diff
@@ -0,0 +1,45 @@
+import { useEffect } from 'react';
+import { useRecoilState } from 'recoil';
+import { searchState, SearchStatus, useGraph } from '@peripleo/peripleo';
+
+/**
+ * The SearchHandler handles one thing: translating a Search
+ * into a list of results by querying the graph.
+ */
+export const ITSBSearchHandler = props => {
+
+  const [search, setSearchState] = useRecoilState(searchState);
+
+  const graph = useGraph();
+
+  useEffect(() => {
+    if (search.status === SearchStatus.PENDING) {
+      const authors = 
+        search.args.filters?.find(f => f.name === 'authors')?.values || [];
+
+      const dateRange = 
+        search.args.filters?.find(f => f.name === 'daterange')?.range;
+      
+      // TODO this could be optimized by filtering authors first!
+      const itineraries = graph.listItineraries(dateRange)
+        .filter(it => authors.includes(it.author));
+
+      setSearchState({
+        args: search.args,
+        status: SearchStatus.OK,
+        result: {
+          total: itineraries.length,
+          items: itineraries
+        }
+      });
+    }
+  }, [search]);
+
+  useEffect(() => {
+    // Trigger initial search
+    setSearchState({ args: {}, status: SearchStatus.PENDING });
+  }, [graph]);
```

Comment:
How do we know for certain that `graph` only changes once, on initialization?

Reply:
Good point... yes, maybe we should re-investigate this to make sure this work generically. In the current implementation, the graph will only get created and mounted once. (I'm using [Promise.all and a derived 'loaded' flag here](https://github.com/performant-software/itsb/blob/peripleo-refactoring/src/App.jsx#L27-L39), which should ensure the Store gets mounted only once; and then [an effect here](https://github.com/performant-software/itsb/blob/peripleo-refactoring/src/components/ITSBStore/ITSBGraphProvider.jsx#L10-L18) that ensures the graph only gets created if the data changes.